### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by AckslD**
A test framework where the sending and reading of messages are mocked so we can simply test what the CQCConnection is sending based on what commands are applied.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+def mock_socket(monkeypatch):
+    def get_mocked_socket(*args, **kwargs):
+        mock_socket = MockSocket(*args, **kwargs)
+        return mock_socket
+
+    monkeypatch.setattr(socket, "socket", get_mocked_socket)
+
+
+class MockedFirstMessage:
+    """Mocks the second header returned by CQCConnection.readMessage"""
```
Comment (lgvacanti) [inline tests/conftest.py]: Is this supposed to be **first** header?

File: tests/test_cqcconnection.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    (commands_to_apply_simple_h, get_expected_headers_simple_h),
+])
+def test_commands(commands_to_apply, get_expected_headers, monkeypatch, mock_socket, mock_read_message):
+
+    with CQCConnection("Test", socket_address=('localhost', 8000), use_classical_communication=False) as cqc:
+        commands_to_apply(cqc)
+
+    expected_headers = get_expected_headers()
+
+    commands_send = list(filter(lambda call: call.name == 'send', cqc._s.calls))
```
Comment (lgvacanti) [inline tests/test_cqcconnection.py]: Maybe rename commands_send to commands_sent. I think that'd make more sense.

File: tests/test_cqcconnection.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    assert len(expected_headers) == len(commands_send)
+    for command, expected in zip(commands_send, expected_headers):
+        print(command.args[0])
+        print(expected)
+        print()
+        if expected is not None:
```
Comment (lgvacanti) [inline tests/test_cqcconnection.py:78]: In what situation would expected be None?

File: tests/conftest.py (hunk 1 above)
Comment (AckslD) [reply tests/conftest.py]: Done

File: tests/test_cqcconnection.py (hunk 2 above)
Comment (AckslD) [reply tests/test_cqcconnection.py]: Done

File: tests/test_cqcconnection.py (hunk 3 above)
Comment (AckslD) [reply tests/test_cqcconnection.py:78]: Done

**Pull request merged**